torch/_functorch/_aot_autograd/schemas.py

```diff
@@ -314,6 +314,8 @@ class ViewAndMutationMeta:
     # This list is generated after calling make_runtime_safe().
     traced_tangent_metas: Optional[List[Any]] = None
 
+    traced_tangent_memory_formats: Optional[List[Any]] = None
```

Comment:
A comment on the structure here would be nice. In particular, I think the new type after our discussion is something like this?
```
Optional[List[Union[MemoryFormat, List[MemoryFormat]]]

# for each tangent at index i:
#     if the tangent is a plain tensor, traced_tangent_memory_formats[i] holds the memory format of the tangent that we need to coerce to
#     if the tangent is a subclass, traced_tangent_memory_formats[i] holds a list of memory formats, containing the expected memory format of the subclass **and** all of its inner tensors
```

what order is that flattened list stored in? in-order tree traversal?

Reply:
Yes, will update it.